### PR TITLE
feat!: add windows support to worker fixture

### PR DESCRIPTION
**Pull request by moorec-aws**
### What was the problem/requirement? (What/Why)
Add Windows support to the worker fixture.

### What was the solution? (How)
Add Windows support to the worker fixture.

### What is the impact of this change?
Adds Windows support to the worker fixture.

### How was this change tested?
Ran the e2e tests from the deadline-worker-agent repository and confirmed they passed:
```
hatch run windows-e2e-test
hatch run linux-e2e-test
```

Set `export WORKER_AGENT_WHL_PATH=/home/user/Projects/git/deadline-cloud-worker-agent/dist/deadline_cloud_worker_agent-0.27.0.post1.dev22-py3-none-any.whl`

Confirmed that the dev version was installed on the worker:
```
C:\Windows\system32> pip list
Package                     Version
--------------------------- ------------------
deadline-cloud-worker-agent 0.27.0.post1.dev22
```

Set `export AWS_ENDPOINT_URL_DEADLINE=<custom-endpoint>`

Confirmed that that it failed because the resource ids were not at that endpoint

### Was this change documented?
No

### Is this a breaking change?
Yes

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

**Comment by Sahmwell** [comment]
🤯 I've been waiting for this

**Comment by Sahmwell** [comment]
Wow! I've been waiting for this!

**Comment by Sahmwell** [comment]
Wow, this is awesome! 🤯

**Comment by Sahmwell** [comment]
Woah cool!

File: src/deadline_test_fixtures/deadline/worker.py [hunk 1]
```diff
@@ -316,6 +415,7 @@ def _launch_instance(self, *, s3_files: list[tuple[str, str]] | None = None) ->
             InstanceType="t3.micro",
             IamInstanceProfile={"Name": self.instance_profile_name},
             SubnetId=self.subnet_id,
+            KeyName="windowstesting",
```
Comment (ttblanchard) [inline src/deadline_test_fixtures/deadline/worker.py]: Is a KeyPair required for running the windows tests? It wasn't previously for Linux. Or is this just some left over testing artifact?

File: src/deadline_test_fixtures/fixtures.py [hunk 2]
```diff
@@ -438,7 +445,10 @@ def worker_config(
         with src_path.open(mode="w") as f:
             json.dump(service_model.model, f)
 
-        dst_path = posixpath.join("/tmp", src_path.name)
+        if operating_system == "AL2023":
```
Comment (ttblanchard) [inline src/deadline_test_fixtures/fixtures.py:448]: All this branching seems like an opportunity to create a subclass for each variant. Maybe for future work.

File: src/deadline_test_fixtures/deadline/worker.py (hunk 1 above)
Comment (moorec-aws) [reply src/deadline_test_fixtures/deadline/worker.py]: left over from testing.

**Review by ttblanchard — DISMISSED** [review]

**Review by ttblanchard — APPROVED** [review]

**Review by epmog — APPROVED** [review]

**Pull request merged**